llvm/include/llvm/Support/DebugCounter.h

```diff
@@ -77,18 +90,28 @@ class DebugCounter {
     auto Result = Us.Counters.find(CounterName);
     if (Result != Us.Counters.end()) {
       auto &CounterInfo = Result->second;
-      ++CounterInfo.Count;
+      int64_t CurrCount = CounterInfo.Count++;
```

Comment:
Not your change, but wow, that function is large, and its main body (below the initial isCountingEnabled check) should really go into an Impl method that is defined in the .cpp file.